src/features/syntaxDecorations.ts

```diff
@@ -134,17 +142,20 @@ export const activate = (context: ExtensionContext) => {
         if (timeout) {
           clearTimeout(timeout);
         }
-        timeout = setTimeout(() => updateDecorations(editor), 200);
+        timeout = setTimeout(() => updateDecorations(editor, true), 200);
       };
 
       if (editor !== undefined && event.document === editor.document) {
         triggerUpdateDecorations(editor);
       }
     }),
+    workspace.onDidCloseTextDocument((event) => {
+      delete decors[event.fileName];
+    }),
```

Comment:
Is this enough? Can we still be leaking somehow? Maybe enforce a maximum number of saved `decors` before adding a new one.